Sources/TuistKit/Generator/GeneratorPaths.swift

```diff
@@ -15,13 +34,18 @@ struct GeneratorPaths {
 
```

Comment:
> the root path is unlikely to change during the generation

The path depends on the project and different projects might have different root directories because we support multiple Tuist directories. For that reason I added the look up using the `RootDirectoryLocator`, which optimizes the lookups using the cache.

Reply:
I did a terrible job explaining this 😛 - it's ok we can improve separately if needed

[GeneratorModelLoader.swift#L44](https://github.com/tuist/tuist/blob/master/Sources/TuistKit/Generator/GeneratorModelLoader.swift#L44) creates the `GeneratorPaths` struct per project load as such it could be tweaked:

```swift
 func loadProject(at path: AbsolutePath) throws -> TuistCore.Project {
      let manifest = try manifestLoader.loadProject(at: path)
      let rootPath = RootDirectoryLocator.shared.locate(from: path)
      let generatorPaths = GeneratorPaths(manifestDirectory: path, rootPath: rootPath)
     // ...
```

That way inside we don't need to do lookups per `resolve` (true it's cached under the hood, however not needed).

Was a minor comment - no worries